.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,45 @@
+name: golangci-lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
+      - 8.0
+      - 8.1
+      - 7.17
+  pull_request:
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  pull-requests: read
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      # Uses Go version from the repository.
+      - name: Read .go-version file
+        id: goversion
+        run: echo "::set-output name=version::$(cat .go-version)"
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: "${{ steps.goversion.outputs.version }}"
```

Comment:
What do you think with this other approach?

https://github.com/elastic/beats/blob/0def363871565f6024f034965fa2a26fa769e51e/.github/workflows/macos-auditbeat.yml#L21-L25

Reply:
I prefer this way, it feels more github action way.